File: csharp/Concepts/Variables.cs
namespace ProgrammingFundamentals.Concepts
{
    public class Variables
    {
        public static void Demonstrate()
        {
            Console.WriteLine("\n=== Variables and Data Types ===");
            string name = "John Doe";     // String type
            int age = 25;                 // Integer type
            double height = 1.75;         // Double type
            bool isStudent = true;        // Boolean type

            Console.WriteLine($"Name: {name} (Type: {name.GetType()})");
            Console.WriteLine($"Age: {age} (Type: {age.GetType()})");
            Console.WriteLine($"Height: {height} (Type: {height.GetType()})");
            Console.WriteLine($"Is Student: {isStudent} (Type: {isStudent.GetType()})");
        }
    }
}

File: java/concepts/Variables.java
package concepts;

public class Variables {
    public static void demonstrate() {
        System.out.println("\n=== Variables and Data Types ===");

        // Types are declared before the variable name.
        String name = "John Doe"; // String is a class
        int age = 25;             // int is a primitive type
        double height = 1.75;     // double is a primitive type for decimals
        boolean isStudent = true; // boolean is a primitive type

        // We can get the class name for object types.
        System.out.println("Name: " + name + " (Type: " + name.getClass().getSimpleName() + ")");
        // For primitives, we state the known type.
        System.out.println("Age: " + age + " (Type: int)");
        System.out.println("Height: " + height + " (Type: double)");
        System.out.println("Is Student: " + isStudent + " (Type: boolean)");
    }
}
